canvas_modules/common-canvas/src/tooltip/tooltip.jsx

```diff
@@ -302,21 +313,44 @@ class ToolTip extends React.Component {
 				((tooltipTop + tooltip.offsetHeight) > document.documentElement.clientHeight)); // to the bottom
 	}
 
+	toggleTooltipOnClick(evt) {
+		// 'blur' event occurs before 'click' event. Because of this, onBlur function is called which hides the tooltip.
+		// To prevent this default behavior, stopPropagation and preventDefault is used.
+		evt.stopPropagation();
+		evt.preventDefault();
+		if (this.state.isTooltipVisible) {
+			// Tooltip is visible and user clicks on trigger element again, hide tooltip
+			this.setTooltipVisible(false);
+		} else {
+			this.setTooltipVisible(true);
+		}
+	}
+
 	render() {
 		let tooltipContent = null;
 		let triggerContent = null;
 		if (this.props.children) {
 			// when children are passed in, tooltip will handle show/hide, otherwise consumer has to hide show/hide tooltip
-			const mouseover = () => this.showTooltipWithDelay();
+			const mouseover = () => this.setTooltipVisible(true);
 			const mouseleave = () => this.setTooltipVisible(false);
 			const mousedown = () => this.setTooltipVisible(false);
+			// `focus` event occurs before `click`. Adding timeout in onFocus function to ensure click is executed first.
+			// Ref - https://stackoverflow.com/a/49512400
+			const onFocus = () => this.showTooltipWithDelay();
+			const onBlur = () => this.setTooltipVisible(false);
+			const click = (evt) => this.toggleTooltipOnClick(evt);
 
+			// If showToolTipOnClick enabled, don't show tooltip on mouseover and mouseleave
```

Comment:
Comments like this aren't really necessary since it just explains what is clear from the code anyway.